realm/realm-library/src/androidTest/java/io/realm/RealmCollectionTests.java

```diff
@@ -181,6 +214,18 @@ public void contains_null() {
         assertFalse(collection.contains(null));
     }
 
+    // Test that the custom equal methods is being used when testing if an object is part of the
+    // collection
+    @Test
+    public void contains_customEqualMethod() {
+        RealmCollection<CustomMethods> collection = createCustomMethodsCollection(realm, collectionClass);
+        // This custom equals method will only consider the field `name` when comparing objects.
+        // So this unmanaged version should be equal to any object with the same value, managed
+        // or not.
+        CustomMethods unmanaged = new CustomMethods("Child0");
```

Comment:
Only test first element? Maybe testing `Child + TEST_SIZE - 1` too (checking the iterators)?